portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -86,12 +72,59 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         response.sendRedirect("/contact.html");
     }
 
+    /** Returns comment limit value from the request, or -1 if user selected 'all' */
+    private int getCommentLimit(HttpServletRequest request) {
+        String limit_string = request.getParameter("comment-limit");
+        int comment_limit = 0;
+
+        try {
+            comment_limit = Integer.parseInt(limit_string);
+        } catch (NumberFormatException n) {
+            if (limit_string.equals("all")) {
```

Comment:
If you end up keeping "all" as a valid parameter value, then I would handle this (and any other special values) separately before trying to parse the number.

That would leave a failure to parse the number as a true exception, rather than just a sign that maybe there is some special handling needed. It is generally good to keep exceptions limited to situations that you don't expect. (But this is not a hard-and-fast rule).